backend/dataall/modules/worksheets/services/worksheet_query_result_service.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import io
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from openpyxl import Workbook
+
+from dataall.base.aws.s3_client import S3_client
+from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.modules.worksheets.db.worksheet_models import WorksheetQueryResult
+from dataall.modules.worksheets.db.worksheet_repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_service import WorksheetService
+
+if TYPE_CHECKING:
+    try:
+        from sqlalchemy.orm import Session
+        from mypy_boto3_s3.client import S3Client
+    except ImportError:
+        print('skipping type checks as stubs are not installed')
+        S3Client = None
+        Session = None
+
+
+class WorksheetQueryResultService:
+    SupportedFormats = {'csv', 'xlsx'}
+
+    @staticmethod
+    def validate_input(data):
+        if not data:
+            raise exceptions.InvalidInput('data', data, 'input is required')
+        if not data.get('athenaQueryId'):
+            raise exceptions.RequiredParameter('athenaQueryId')
+        if not data.get('fileFormat'):
+            raise exceptions.RequiredParameter('fileFormat')
+        if data.get('fileFormat', '').lower() not in WorksheetQueryResultService.SupportedFormats:
+            raise exceptions.InvalidInput(
+                'fileFormat', data.get('fileFormat'), ', '.join(WorksheetQueryResultService.SupportedFormats)
+            )
+
+    @staticmethod
+    def get_output_bucket(session: 'Session', environment_uri: str) -> str:
+        environment = EnvironmentService.get_environment_by_uri(session, environment_uri)
+        bucket = environment.EnvironmentDefaultBucketName
+        return bucket
+
+    @staticmethod
+    def create_query_result(
+        environment_bucket: str, athena_workgroup: str, worksheet_uri: str, region: str, aws_account_id: str, data: dict
+    ) -> WorksheetQueryResult:
+        sql_query_result = WorksheetQueryResult(
+            worksheetUri=worksheet_uri,
+            AthenaQueryId=data.get('athenaQueryId'),
+            fileFormat=data.get('fileFormat'),
+            OutputLocation=f's3://{environment_bucket}/athenaqueries/{athena_workgroup}/',
+            region=region,
+            AwsAccountId=aws_account_id,
+            queryType='data',
+        )
+        return sql_query_result
+
```

Comment:
nit: in line 65 use Enum value instead of directly writing `csv` and for the athena_queries_dir we could set a constant at the top level of the file, and re-use it
```
ATHENA_QUERIES_PATH = 'athenaqueries'

class WorksheetQueryResultService:
    @staticmethod
    def get_environment_bucket(environment) -> str:
```

Reply:
changes are done